File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/profissional/dto/ProfissionalDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.profissional.dto;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class ProfissionalDTO implements DTO {

    private UUID id;

    private UUID pessoaId;
    private String pessoaNome;

    private String conselho;
    private String codigoConselho;
    private String tipoRemuneracao;

    private String banco;
    private String conta;
    private String chavePix;

    private Boolean ativo;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}

File: src/frontend/src/app/components/atendimento/profissional/model/profissional-dto.ts
import {
  Exclude,
  Expose,
  Transform,
  TransformationType,
  TransformFnParams,
} from 'class-transformer';
import { TipoRemuneracao } from './tipo-remuneracao.enum';
import { UF } from './uf.enum';

@Exclude()
export class ProfissionalDTO {
  @Expose() id?: string;
  @Expose() pessoaId?: string;
  @Expose() pessoaNome?: string;
  @Expose() conselho?: string;
  @Expose() codigoConselho?: string;

  @Transform((params: TransformFnParams) => {
    const { type, value } = params;
    if (TransformationType.PLAIN_TO_CLASS === type) {
      return typeof value === 'string' ? TipoRemuneracao.getByKey(value) : value;
    }
    if (TransformationType.CLASS_TO_PLAIN === type) {
      if (value && typeof value.getKey === 'function') return value.getKey();
      if (typeof value === 'string') return value;
      return value;
    }
    return value;
  })
  @Expose()
  tipoRemuneracao?: TipoRemuneracao;

  @Expose() banco?: string;
  @Expose() conta?: string;
  @Expose() chavePix?: string;

  @Transform((params: TransformFnParams) => {
    const { type, value } = params;
    if (TransformationType.PLAIN_TO_CLASS === type) {
      return typeof value === 'string' ? UF.getByKey(value) : value;
    }
    if (TransformationType.CLASS_TO_PLAIN === type) {
      if (value && typeof value.getKey === 'function') return value.getKey();
      if (typeof value === 'string') return value;
      return value;
    }
    return value;
  })
  @Expose()
  uf?: UF;

  @Expose() ativo?: boolean;
  @Expose() createdAt?: Date;
  @Expose() updatedAt?: Date;
  @Expose() createdBy?: string;
  @Expose() updatedBy?: string;
}
